plugins/repository-hdfs/build.gradle

```diff
@@ -230,6 +230,11 @@ if (Os.isFamily(Os.FAMILY_WINDOWS)) {
   fixtureSupported = true
 }
 
+boolean legalPath = rootProject.rootDir.toString().contains(" ") == false
```

Comment:
I'm not sure a regular space in the path is what was causing the original issue. The issue was because a space character was _HTTP encoded_ to `%20` (and then encoded again to `%2520`). The `%` sign with numbers trips up string formatting in Java, which is what caused the exception.

Perhaps it might make sense to try formatting the string the same way HDFS does in a try block, catch the exception and mark the path illegal in that case.

Reply:
The offending code in HDFS does:

```java
ThreadFactory workerFactory = new ThreadFactoryBuilder()
        .setDaemon(true)
        .setNameFormat("FsVolumeImplWorker-" + parent.toString() + "-%d") // <----
        .build();
```
Which checks the name format eagerly to make sure it works with:
```java
public ThreadFactoryBuilder setNameFormat(String nameFormat) {
    String.format(nameFormat, 0); // fail fast if the format is bad or null
    this.nameFormat = nameFormat;
    return this;
  }
```

So perhaps we could try formatting the string the exact same way HDFS does to ensure max compatibility in this regard.
```groovy
try {
    // See o.a.h.h.s.d.f.i.FsVolumeImpl#initializeCacheExecutor:150
    String.format(rootProject.rootDir.toString() + '-%d', 0) 
} catch (IllegalFormatException ife) {
    legalPath = false;
}
```

Reply:
Ok, the URL encoding hasn't happened by this point, so looking for a literal space finds the issue here:

```
$ pwd 
/var/lib/jenkins/workspace/elastic+elasticsearch+6.x+multijob-unix-compatibility/os/ubuntu && virtual
$ pwd | xxd
00000000: 2f76 6172 2f6c 6962 2f6a 656e 6b69 6e73  /var/lib/jenkins
00000010: 2f77 6f72 6b73 7061 6365 2f65 6c61 7374  /workspace/elast
00000020: 6963 2b65 6c61 7374 6963 7365 6172 6368  ic+elasticsearch
00000030: 2b36 2e78 2b6d 756c 7469 6a6f 622d 756e  +6.x+multijob-un
00000040: 6978 2d63 6f6d 7061 7469 6269 6c69 7479  ix-compatibility
00000050: 2f6f 732f 7562 756e 7475 2026 2620 7669  /os/ubuntu && vi
00000060: 7274 7561 6c0a                           rtual.
```

However, from the code you've shown it looks like any path that contains anything that needs URL-encoding is going to cause issues here - `%` for instance - so I agree that trying to catch an `IllegalFormatException` seems like it'll be more faithful. Can you dig out how the URL-encoding is happening (e.g. does it use `java.net.URLEncoder` (twice?!) or is it more complicated)?

Reply:
That's a good question. I'll take a look but I wouldn't be surprised if its something deep in Hadoop or far away from where it's breaking. If it's not encoded at this point the space check should be fine enough then. Thanks!